src/util/sock.cpp

```diff
@@ -39,27 +39,28 @@ Sock::Sock(Sock&& other)
     other.m_socket = INVALID_SOCKET;
 }
 
-Sock::~Sock() { Reset(); }
+Sock::~Sock() { *this = Sock{INVALID_SOCKET}; }
```

Comment:
Assigning *this in the descriptor looks a bit strange to me. Let's be sure that this isn't undefined behavior or footguns of some kind.

Reply:
Oops! Yes, sounds like a more straightforwardly safe solution to me. Thanks.